python/cudf/cudf/bindings/stream_compaction.pyx

```diff
@@ -89,23 +100,44 @@ def apply_apply_boolean_mask(cols, mask):
 
 
 def apply_drop_nulls(cols, how="any", subset=None, thresh=None):
+    """
+    Drops null rows from cols. 
+
+    Parameters
+    ----------
+    cols : List of Columns
+    how  : "any" or "all". If thresh is None, drops rows of cols that have any
+           nulls or all nulls (respectively) in subset (default: "any")
+    subset : List of Columns. If set, then these columns are checked for nulls
+             rather than all of cols (optional)
+    thresh : Minimum number of non-nulls required to keep a row (optional)
+
+    Returns
+    -------
+    List of Columns
+    """
     cdef cudf_table c_out_table
     cdef cudf_table* c_in_table = table_from_columns(cols)
+
+    # if subset is None, we use cols as keys
+    # if subset is empty, we pass an empty keys table, which will cause
+    # cudf::drop_nulls() to return a copy of the input table
     cdef cudf_table* c_keys_table = (table_from_columns(cols) if subset is None
                                      else table_from_columns(subset))
 
-    cdef any_or_all drop_if
-    cdef gdf_size_type keep_threshold = len(cols)
+    # default: "any" means threshold should be number of key columns
+    cdef gdf_size_type c_keep_threshold = (len(cols) if subset is None
+                                           else len(subset))
 
-    # Use threshold if specified, otherwise set it based on how
+    # Use `thresh` if specified, otherwise set threshold based on `how`
     if thresh:
```

Comment:
This should be `if thresh is not None`.

`0` is considered `False` so if `thresh` is specified `0`, we fail to update `c_keep_threshold`.

Reply:
Right, None quacks like a False.